packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -206,7 +206,7 @@ abstract class WidgetsBinding extends BindingBase implements GestureBinding, Ren
     buildOwner.finalizeTree();
     // TODO(ianh): Following code should not be included in release mode, only profile and debug modes.
     if (_needToReportFirstFrame) {
-      if (_thisFrameWasUseful) {
+      if (!_thisFrameWasUseful) {
         _thisFrameWasUseful = true;
```

Comment:
rather than negating the condition, switch the two branches around.

Also, oops. :-|